### DISCUSSION TITLE
Should we use ResultOf on fragment documents?

### DISCUSSION THREAD

**Discussion by khurram-ansari** [0no-co/gql.tada#309] [Q&A] [answered]
Let's say I just want to use a certain type in helper function but not in component.

Here is a **fragment doc**:

```
export const CountryStateFragmentDocument = graphql(/* GraphQL */ `
  fragment CountryState on CountryState {
    code
    name
  }
`);
```

One way I can get the type without using `readFragment` is by adding `@unmask` on the fragment and using `FragmentOf`.

Other one is using `ResultOf<typeof CountryStateFragmentDocument>`

So, are these both approaches the same and should we use it like this way?

**Comment by kitten** [comment, marked as answer]
There's more information here: https://gql-tada.0no.co/guides/fragment-colocation#fragment-colocation-1

Whether you're using `@_unmask` or not, the `FragmentOf` utility type will continue to work and give you the correct type. You _can_ use `ResultOf` on an unmasked fragment, but if you ever remove `@_unmask`, it'll stop working, whereas `readFragment` and `FragmentOf` are designed to work whether the fragment is unmasked or not, to allow you to change your mind later on.

**Reply by khurram-ansari** [reply]
Makes sense, thank you so much.
